sdk/communication/azure-communication-identity/azure/communication/identity/_shared/policy.py

```diff
@@ -52,7 +60,38 @@ def _sign_request(self, request):
         verb = request.http_request.method.upper()
 
         # Get the path and query from url, which looks like https://host/path/query
-        query_url = str(request.http_request.url[len(self._host) + 8 :])
+        parsed_url: ParseResult = urlparse(request.http_request.url)
+        query_url = parsed_url.path
+
+        if parsed_url.query:
+            query_url += "?" + parsed_url.query
+
+        # Need URL() to get a correct encoded key value, from "%3A" to ":", when transport is in type AioHttpTransport.
+        # There's a similar scenario in azure-storage-blob and azure-appconfiguration, the check logic is from there.
```

Comment:
Is this a bug fix that should be documented on the Changelog?

Reply:
I copied this from jobrouter library, to keep them in sync. (_shared folder must be in sync across all packages).
Once my PR would gets merged, I was going to propagate all the _shared folder changes in other communication packages.